src/gosling-genomic-axis/axis-track.ts

```diff
@@ -8,35 +8,134 @@ import { format, precisionPrefix, formatPrefix } from 'd3-format';
 
-function AxisTrack(HGC: typeof import('@higlass/available-for-plugins'), ...args: any[]): any {
-    if (!new.target) {
-        throw new Error('Uncaught TypeError: Class constructor cannot be invoked without "new"');
+const config: TrackConfig<AxisTrackOptions> = {
+    type: 'axis-track',
+    datatype: ['multivec', 'epilogos'],
+    local: false,
+    orientation: '1d-horizontal',
+    thumbnail: new DOMParser().parseFromString(icon, 'text/xml').documentElement,
+    defaultOptions: {
+        innerRadius: 340,
+        outerRadius: 310,
+        startAngle: 0,
+        endAngle: 360,
+        width: 700,
+        height: 700,
+        layout: 'linear',
+        labelPosition: 'none',
+        labelColor: 'black',
+        labelTextOpacity: 0.4,
+        trackBorderWidth: 0,
+        trackBorderColor: 'black',
+        tickPositions: 'even',
+        fontSize: 12,
+        fontFamily: 'sans-serif', // 'Arial',
+        fontWeight: 'normal',
+        color: '#808080',
+        stroke: '#ffffff',
+        backgroundColor: 'transparent',
+        showMousePosition: false,
+        tickColor: TICK_COLOR
     }
+};
 
+const factory: PluginTrackFactory<AxisTrackOptions> = (HGC, context, options) => {
     const { absToChr, colorToHex, pixiTextToSvg, svgLine, showMousePosition } = HGC.utils;
 
-    class AxisTrackClass extends HGC.tracks.PixiTrack {
+    function createTickText(text: string, style: Partial<PIXI.ITextStyle>): TickText {
+        return Object.assign(new HGC.libraries.PIXI.Text(text, style), { hashValue: Math.random() });
+    }
+
+    class AxisTrackClass extends HGC.tracks.PixiTrack<typeof options> {
         allTexts: TickLabelInfo[];
+        searchField: null;
+        chromInfo: ChromInfo;
+        dataConfig: Record<string, unknown>;
+        pTicksCircular: PIXI.Graphics;
```

Comment:
Given that we have explicit types, we can gradually revisit each file in the future and try to directly use HGC libraries, e.g., `HGC.libraries.PIXI.Graphics`.

Reply:
Since types are erased, it technically shouldn't matter where they are imported but I agree for consistency this would be nice!